### PR TITLE
Postfixes ipc enhanced

### PR DESCRIPTION
**Pull request by jbbjarnason**
* fix regex in dbus name throwing.
* split implementation of el1xx and el2xx to source file.
* split impl of dbus_server_iface_mock to source file. (why is not call it dbus_client_iface_mock ???)
* cleanup includes in touched code

Debug binaries gain size a bit, 
```bash
jonb@jonb-work-desktop ~/P/f/b/ci-clang-debug (postfixes-ipc-enhanced)> du -h exes/tfcctl/tfcctl 
45M     exes/tfcctl/tfcctl
jonb@jonb-work-desktop ~/P/f/b/ci-clang-debug (postfixes-ipc-enhanced)> du -h exes/ethercat/ethercat 
45M     exes/ethercat/ethercat
# versus
jonb@jonb-work-desktop ~/P/f/b/ci-clang-debug (main)> du -h exes/tfcctl/tfcctl
42M	exes/tfcctl/tfcctl
jonb@jonb-work-desktop ~/P/f/b/ci-clang-debug (main)> du -h exes/ethercat/ethercat
41M	exes/ethercat/ethercat
```

Release binaries are reduced a tiny bit
```bash
jonb@jonb-work-desktop ~/P/f/b/ci-clang-release (postfixes-ipc-enhanced)> du -h exes/ethercat/ethercat
13M     exes/ethercat/ethercat
jonb@jonb-work-desktop ~/P/f/b/ci-clang-release (postfixes-ipc-enhanced)> du -h exes/tfcctl/tfcctl
12M     exes/tfcctl/tfcctl
# versus
jonb@jonb-work-desktop ~/P/f/b/ci-clang-release (main)> du -h exes/ethercat/ethercat
13M	exes/ethercat/ethercat
jonb@jonb-work-desktop ~/P/f/b/ci-clang-release (main)> du -h exes/tfcctl/tfcctl
13M	exes/tfcctl/tfcctl
```

### CODE REVIEW

File: exes/ethercat/tests/devices/beckhoff.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    expect(slot_1.value() == true);
+    expect(slot_2.value() == true);
+
+    // the above lines could be replaced with:
+    // EXPECT_CALL(vars.device.transmitters().at(0), async_send(true, _));
+    // EXPECT_CALL(vars.device.transmitters().at(1), async_send(true, _));
+    // vars.device.process_data(buffer, {});
+    // I think we should do that, but it requires manipulating private and make gmock of signal class.
+    // Would make it unnecessary to run the ctx for a while, and would make the test more deterministic.
+    // todo thoughts???
```
Comment (jbbjarnason) [inline exes/ethercat/tests/devices/beckhoff.cpp:62]: Elaborate your thoughts on this :smile:

File: exes/ethercat/src/devices/beckhoff.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
+#include <fmt/format.h>
+
+#include <tfc/ec/devices/beckhoff/EL1xxx.hpp>
+#include <tfc/ec/devices/beckhoff/EL2xxx.hpp>
+#include <tfc/ipc.hpp>
+#include <tfc/ipc/details/dbus_server_iface.hpp>
+
+namespace tfc::ec::devices::beckhoff {
+
+// TODO discuss! I would like to remove the name of the device from the ipc channel name
```
Comment (jbbjarnason) [inline exes/ethercat/src/devices/beckhoff.cpp:10]: Elaborate your thoughts on this :smile:

**Comment by jbbjarnason** [comment]
Will make a bit more refactoring, in terms of testing will look better

File: exes/ethercat/tests/devices/beckhoff.cpp (hunk 1 above)
Comment (jbbjarnason) [reply exes/ethercat/tests/devices/beckhoff.cpp]: The tests where flaky so I made this change and made the test better

File: exes/ethercat/src/devices/beckhoff.cpp (hunk 2 above)
Comment (omarhogni) [reply exes/ethercat/src/devices/beckhoff.cpp]: How would you name them?
Are you thinking about a per network digital IO list like plc's do
```
Q0.1
Q0.2
```

I don't really like that idea, then if you switch to something generic like 
```
DI16.{ethercat_id}.0
DI16.{ethercat_id}.1
```

**Review by omarhogni — APPROVED** [review]

**Comment by omarhogni** [comment]
@jbbjarnason have you done pre and post change build time comparisons?

**Comment by omarhogni** [comment]
> @jbbjarnason have you done pre and post change build time comparisons?

Based on the github pipelines on main vs, this branch it looks like this makes about 30 minutes of difference. 74% reduction. Pretty sweet.

**Comment by jbbjarnason** [comment]
> > @jbbjarnason have you done pre and post change build time comparisons?
> 
> Based on the github pipelines on main vs, this branch it looks like this makes about 30 minutes of difference. 77% reduction. Pretty sweet.

It might be a bit biased since the vcpkg cache was updating using fmt from vcpkg repo. 
But 12 minutes ish is not bad.

File: exes/ethercat/src/devices/beckhoff.cpp (hunk 2 above)
Comment (jbbjarnason) [reply exes/ethercat/src/devices/beckhoff.cpp]: What I am thinking is something like `ethercat.def.slave3.input4` but let's discuss this in person. Will remove comment

**Pull request merged**